### PR TITLE
BarChart panel: Add support for Tooltip in BarChartPanel

### LINKED ISSUE #33848: BarChart: Enable Tooltip support

### PR DESCRIPTION
**Pull request by dprokop**
Closes https://github.com/grafana/grafana/issues/33848

TODO
- [x] Move BarChart to core (https://github.com/grafana/grafana/pull/33947)

### CODE REVIEW

File: packages/grafana-ui/src/components/BarChart/bars.ts [hunk 1]
```diff
@@ -144,17 +126,14 @@ export function getConfig(opts: BarsOptions) {
                 let lft = Math.round(xOff + (_dir === 1 ? x0 : xDim - x0 - wid));
                 let barWid = Math.round(wid);
 
-                // prettier-ignore
                 if (dataY[ix] != null) {
                   let yPos = valToPosY(dataY[ix]!, scaleY, yDim, yOff);
 
-                  /* eslint-disable no-multi-spaces */
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/BarChart/bars.ts:151]: @dprokop, c'mon, this was explicitly aligned this way on purpose.

File: packages/grafana-ui/src/components/BarChart/bars.ts [hunk 2]
```diff
@@ -1,13 +1,10 @@
-import uPlot, { Axis, Series, Cursor, Select } from 'uplot';
+import uPlot, { Axis, Series } from 'uplot';
 import { Quadtree, Rect, pointWithin } from './quadtree';
 import { distribute, SPACE_BETWEEN } from './distribute';
+import { TooltipInterpolator } from '../uPlot/types';
+import { ScaleDirection, ScaleOrientation } from '../uPlot/config';
 
-const pxRatio = devicePixelRatio;
-
-const groupDistr = SPACE_BETWEEN;
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/BarChart/bars.ts:7]: not sure why this needs to be removed. the whole point was to make it easy to change in one place.

File: packages/grafana-ui/src/components/BarChart/bars.ts [hunk 3]
```diff
@@ -1,13 +1,10 @@
...
-const barDistr = SPACE_BETWEEN;
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/BarChart/bars.ts:8]: not sure why these need to be removed. the whole point was to make it easy to change in one place.

File: packages/grafana-ui/src/components/BarChart/bars.ts [hunk 4]
```diff
@@ -235,50 +190,48 @@ export function getConfig(opts: BarsOptions) {
...
+      qt.get(cx, cy, 1, 1, (o) => {
+        if (pointWithin(cx, cy, o.x, o.y, o.x + o.w, o.y + o.h)) {
+          found = o;
+        }
+      });
 
-    if (found) {
-      // prettier-ignore
-      if (found !== hovered) {
-        /* eslint-disable no-multi-spaces */
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/BarChart/bars.ts:252]: same comment as the other one. this is not better, imo.

Comment (dprokop) [reply packages/grafana-ui/src/components/BarChart/bars.ts:252]: yeah, u r right, it's me mindlessly removing ignores :)

File: packages/grafana-ui/src/components/BarChart/bars.ts (hunk 3 above)
Comment (leeoniya) [reply packages/grafana-ui/src/components/BarChart/bars.ts:8]: SPACE_BETWEEN has no semantic meaning, but barDistr does. it's the same reason you may want `declare type Spacing: number`. even though you can simply use `: number` inline everywhere.

**Comment by dprokop** [comment]
> a lot of this diff is changes for the sake of formatting and in most cases the new formatting is harder to read/audit due to misalignment :(

Not that much if you look closer at the [revert diff ](68e9e57626e5ec312ee2309ef9669f1aacc81858) @leeoniya  ;)

**Comment by leeoniya** [comment]
sorry, must have been looking at compressed partial diff. will review in a bit, thanks!

**Review by ryantxu — APPROVED** [review]
works as expected

**Pull request merged**